packages/apollo-engine-reporting/src/__tests__/extension.test.ts

```diff
@@ -248,91 +262,185 @@ describe('variableJson output for sendVariableValues transform: custom function
+  it("expect stringify(null) for values set to null by custom modifier", () => {
     expect(
       makeTraceDetails(variables, { transform: modifier }).variablesJson[
         secondKey
-      ],
+        ]
     ).toEqual(JSON.stringify(null));
   });
 
-  const errorThrowingModifier = (input: {
+  const errorThrowingModifier = (_input: {
     variables: Record<string, any>;
   }): Record<string, any> => {
-    throw new GraphQLError('testing error handling');
+    throw new GraphQLError("testing error handling");
   };
 
-  it('redact all variable values when custom modifier throws an error', () => {
+  it("redact all variable values when custom modifier throws an error", () => {
     const variableJson = makeTraceDetails(variables, {
-      transform: errorThrowingModifier,
+      transform: errorThrowingModifier
     }).variablesJson;
     Object.keys(variableJson).forEach(variableName => {
       expect(variableJson[variableName]).toEqual(
-        JSON.stringify('[PREDICATE_FUNCTION_ERROR]'),
+        JSON.stringify("[PREDICATE_FUNCTION_ERROR]")
       );
     });
     expect(Object.keys(variableJson).sort()).toEqual(
-      Object.keys(variables).sort(),
+      Object.keys(variables).sort()
     );
   });
 });
 
-describe('Catch circular reference error during JSON.stringify', () => {
+describe("Catch circular reference error during JSON.stringify", () => {
   interface SelfCircular {
     self?: SelfCircular;
   }
 
   const circularReference: SelfCircular = {};
-  circularReference['self'] = circularReference;
+  circularReference["self"] = circularReference;
 
   const circularVariables = {
-    bad: circularReference,
+    bad: circularReference
   };
 
   expect(
-    makeTraceDetails(circularVariables, { all: true }).variablesJson['bad'],
-  ).toEqual(JSON.stringify('[Unable to convert value to JSON]'));
+    makeTraceDetails(circularVariables, { all: true }).variablesJson["bad"]
+  ).toEqual(JSON.stringify("[Unable to convert value to JSON]"));
 });
 
 function makeTestHTTP(): Trace.HTTP {
   return new Trace.HTTP({
     method: Trace.HTTP.Method.UNKNOWN,
     host: null,
-    path: null,
+    path: null
   });
 }
 
 /**
  * TESTS FOR THE sendHeaders REPORTING OPTION
  */
 const headers = new Headers();
-headers.append('name', 'value');
-headers.append('authorization', 'blahblah'); // THIS SHOULD NEVER BE SENT
+headers.append("name", "value");
+headers.append("authorization", "blahblah"); // THIS SHOULD NEVER BE SENT
+
+const headersOutput = { name: new Trace.HTTP.Values({ value: ["value"] }) };
+
+describe("tests for the shouldReportQuery reporting option", () => {
+  it("report no traces", async () => {
+    const schema = makeExecutableSchema({ typeDefs });
+    addMockFunctionsToSchema({ schema });
+    enableGraphQLExtensions(schema);
+
+    async function addTrace(_args: AddTraceArgs) {
+      throw new Error("Should not add any traces");
+    }
```

Comment:
A more common approach in javascript / jest testing would be to make this a mock function and create a / some expectations around that mock after the test has mostly run. In this case:

```js
// create a mock function - this in particular does nothing
const addTrace = jest.fn();
// ...

/* Use the `toBeCalled` or one of its variations to ensure the mock fn was never called
 * i.e. `.not.toBeCalled()`, `.toBeCalledTimes(0)`.
 *
 * This expectation should happen at the end of the test, or at the location where it
 * makes sense. That's to say, this won't be enforced across the entire test by writing
 * the expectation once anywhere.
 */
expect(addTrace).not.toBeCalled();
```